File: frontend/src/types/whatIf.ts
export interface CounterfactualParameters {
  temperature_change?: number;
}

export interface WhatIfRequest {
  investigation_id: number;
  parameters: CounterfactualParameters;
}

export interface WhatIfResponse {
  before_risk: string;
  after_risk: string;
  result: { risk_reduction: number };
}

File: backend/api/schemas/what_if.py
from pydantic import BaseModel

from api.schemas.tools import CounterfactualInputSchema, CounterfactualOutputSchema


class WhatIfRequest(BaseModel):
    investigation_id: int
    parameters: CounterfactualInputSchema


class WhatIfResponse(BaseModel):
    before_risk: str
    after_risk: str
    result: CounterfactualOutputSchema
